<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>The useage of GeoTcgaData</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">The useage of GeoTcgaData</h1>



<hr />
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>GEO and TCGA provide us with a wealth of data, such as RNA-seq, DNA Methylation, and Copy number variation data. It’s easy to download data from TCGA using the gdc tool, but processing these data into a format suitable for bioinformatics analysis requires more work. This R package was developed to handle these data.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">library</span>(GeoTcgaData)</a></code></pre></div>
</div>
<div id="common-operations-on-priogene" class="section level1">
<h1>Common operations on prioGene</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"></a>
<a class="sourceLine" id="cb2-2" title="2"><span class="co">#&#39; This function can find the mean value of the gene in each module.</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-4" title="4"><span class="co">#&#39; param geneExpress a data.frame</span></a>
<a class="sourceLine" id="cb2-5" title="5"><span class="co">#&#39; param module a data.frame</span></a>
<a class="sourceLine" id="cb2-6" title="6"><span class="co">#&#39; param result a string</span></a>
<a class="sourceLine" id="cb2-7" title="7"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-8" title="8"><span class="co">#&#39; @return a matrix, means the mean of gene expression value in</span></a>
<a class="sourceLine" id="cb2-9" title="9"><span class="co">#&#39; the same module</span></a>
<a class="sourceLine" id="cb2-10" title="10">result &lt;-<span class="st"> </span><span class="kw">cal_mean_module</span>(geneExpress,module)</a>
<a class="sourceLine" id="cb2-11" title="11"></a>
<a class="sourceLine" id="cb2-12" title="12"><span class="co">#&#39; Average the expression data of different ids for the same gene in the chip</span></a>
<a class="sourceLine" id="cb2-13" title="13"><span class="co">#&#39; expression profile of GEO or TCGA</span></a>
<a class="sourceLine" id="cb2-14" title="14"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-15" title="15"><span class="co">#&#39; param file1 a data.frame</span></a>
<a class="sourceLine" id="cb2-16" title="16"><span class="co">#&#39; param k a number</span></a>
<a class="sourceLine" id="cb2-17" title="17"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-18" title="18"><span class="co">#&#39; @return a data.frame, the values of same genes in gene expression profile</span></a>
<a class="sourceLine" id="cb2-19" title="19"></a>
<a class="sourceLine" id="cb2-20" title="20"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-21" title="21"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-22" title="22">aa &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Gene Symbol&quot;</span>,<span class="st">&quot;MARCH1&quot;</span>,<span class="st">&quot;MARC1&quot;</span>,<span class="st">&quot;MARCH1&quot;</span>,<span class="st">&quot;MARCH1&quot;</span>,<span class="st">&quot;MARCH1&quot;</span>)</a>
<a class="sourceLine" id="cb2-23" title="23">bb &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GSM1629982&quot;</span>,<span class="st">&quot;2.969058399&quot;</span>,<span class="st">&quot;4.722410064&quot;</span>,<span class="st">&quot;8.165514853&quot;</span>,<span class="st">&quot;8.24243893&quot;</span>,<span class="st">&quot;8.60815086&quot;</span>)</a>
<a class="sourceLine" id="cb2-24" title="24">cc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GSM1629982&quot;</span>,<span class="st">&quot;3.969058399&quot;</span>,<span class="st">&quot;5.722410064&quot;</span>,<span class="st">&quot;7.165514853&quot;</span>,<span class="st">&quot;6.24243893&quot;</span>,<span class="st">&quot;7.60815086&quot;</span>)</a>
<a class="sourceLine" id="cb2-25" title="25">file1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">aa=</span>aa,<span class="dt">bb=</span>bb,<span class="dt">cc=</span>cc)</a>
<a class="sourceLine" id="cb2-26" title="26">result &lt;-<span class="st"> </span><span class="kw">gene_ave</span>(file1)</a>
<a class="sourceLine" id="cb2-27" title="27"><span class="co">#&gt; Warning in gene_ave(file1): 强制改变过程中产生了NA</span></a>
<a class="sourceLine" id="cb2-28" title="28"></a>
<a class="sourceLine" id="cb2-29" title="29"><span class="co">#&gt; Warning in gene_ave(file1): 强制改变过程中产生了NA</span></a>
<a class="sourceLine" id="cb2-30" title="30"></a>
<a class="sourceLine" id="cb2-31" title="31"><span class="co">#&#39; Get the differentially expressioned genes using DESeq2 package </span></a>
<a class="sourceLine" id="cb2-32" title="32"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-33" title="33"><span class="co">#&#39; param profile a data.frame</span></a>
<a class="sourceLine" id="cb2-34" title="34"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-35" title="35"><span class="co">#&#39; @return a data.frame, a intermediate results of DESeq2</span></a>
<a class="sourceLine" id="cb2-36" title="36"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-37" title="37"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-38" title="38">profile2 &lt;-<span class="st"> </span><span class="kw">classify_sample</span>(profile)</a>
<a class="sourceLine" id="cb2-39" title="39"></a>
<a class="sourceLine" id="cb2-40" title="40"></a>
<a class="sourceLine" id="cb2-41" title="41"><span class="co">#&#39; Get the differentially expressioned genes using DESeq2 package</span></a>
<a class="sourceLine" id="cb2-42" title="42"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-43" title="43"><span class="co">#&#39; param profile2 a result of classify_sample</span></a>
<a class="sourceLine" id="cb2-44" title="44"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-45" title="45"><span class="co">#&#39; @return a matrix, information of differential expression genes</span></a>
<a class="sourceLine" id="cb2-46" title="46"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-47" title="47"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-48" title="48">profile2 &lt;-<span class="st"> </span><span class="kw">classify_sample</span>(profile)</a>
<a class="sourceLine" id="cb2-49" title="49">jieguo &lt;-<span class="st"> </span><span class="kw">diff_gene</span>(profile2)</a>
<a class="sourceLine" id="cb2-50" title="50"><span class="co">#&gt; Loading required namespace: DESeq2</span></a>
<a class="sourceLine" id="cb2-51" title="51"><span class="co">#&gt; please install the package DESeq2!</span></a>
<a class="sourceLine" id="cb2-52" title="52"></a>
<a class="sourceLine" id="cb2-53" title="53"><span class="co">#&#39; Merge methylation data downloaded from TCGA</span></a>
<a class="sourceLine" id="cb2-54" title="54"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-55" title="55"><span class="co">#&#39; param dirr a string for the directory of methylation data download from tcga</span></a>
<a class="sourceLine" id="cb2-56" title="56"><span class="co">#&#39; useing the tools gdc</span></a>
<a class="sourceLine" id="cb2-57" title="57"><span class="co">#&#39; @return a matrix, a combined methylation expression spectrum matrix</span></a>
<a class="sourceLine" id="cb2-58" title="58"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-59" title="59"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-60" title="60">mearge_result &lt;-<span class="st"> </span><span class="kw">Merge_methy_tcga</span>(<span class="kw">system.file</span>(<span class="kw">file.path</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;methy&quot;</span>),<span class="dt">package=</span><span class="st">&quot;GeoTcgaData&quot;</span>))</a>
<a class="sourceLine" id="cb2-61" title="61"><span class="co">#&gt; Warning in gene_ave(rep1_result): 强制改变过程中产生了NA</span></a>
<a class="sourceLine" id="cb2-62" title="62"><span class="co">#&gt; Warning in gene_ave(rep1_result): 强制改变过程中产生了NA</span></a>
<a class="sourceLine" id="cb2-63" title="63"></a>
<a class="sourceLine" id="cb2-64" title="64"><span class="co">#&#39; Combine clinical information obtained from TCGA and extract survival data</span></a>
<a class="sourceLine" id="cb2-65" title="65"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-66" title="66"><span class="co">#&#39; param Files_dir1 a dir data</span></a>
<a class="sourceLine" id="cb2-67" title="67"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-68" title="68"><span class="co">#&#39; @return a matrix, survival time and survival state in TCGA</span></a>
<a class="sourceLine" id="cb2-69" title="69"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-70" title="70"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-71" title="71"><span class="kw">tcga_cli_deal</span>(<span class="kw">system.file</span>(<span class="kw">file.path</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;tcga_cli&quot;</span>),<span class="dt">package=</span><span class="st">&quot;GeoTcgaData&quot;</span>))</a>
<a class="sourceLine" id="cb2-72" title="72"><span class="co">#&gt; Warning in utils::read.table(file.path(Files_dir1, file), sep = &quot;\t&quot;,</span></a>
<a class="sourceLine" id="cb2-73" title="73"><span class="co">#&gt; header = T): incomplete final line found by readTableHeader on &#39;C:/</span></a>
<a class="sourceLine" id="cb2-74" title="74"><span class="co">#&gt; Users/Administrator.MS-20180914RLKI/AppData/Local/Temp/RtmpWK9Cqj/</span></a>
<a class="sourceLine" id="cb2-75" title="75"><span class="co">#&gt; Rinst3c9c32f11d2a/GeoTcgaData/extdata/tcga_cli/at.orl.TCGA-2V-A95S.xml&#39;</span></a>
<a class="sourceLine" id="cb2-76" title="76"><span class="co">#&gt; Warning in utils::read.table(file.path(Files_dir1, file), sep = &quot;\t&quot;,</span></a>
<a class="sourceLine" id="cb2-77" title="77"><span class="co">#&gt; header = T): incomplete final line found by readTableHeader on &#39;C:/</span></a>
<a class="sourceLine" id="cb2-78" title="78"><span class="co">#&gt; Users/Administrator.MS-20180914RLKI/AppData/Local/Temp/RtmpWK9Cqj/</span></a>
<a class="sourceLine" id="cb2-79" title="79"><span class="co">#&gt; Rinst3c9c32f11d2a/GeoTcgaData/extdata/tcga_cli/ati.org.TCGA-2Y-A9GT.xml&#39;</span></a>
<a class="sourceLine" id="cb2-80" title="80"><span class="co">#&gt; Warning in utils::read.table(file.path(Files_dir1, file), sep = &quot;\t&quot;,</span></a>
<a class="sourceLine" id="cb2-81" title="81"><span class="co">#&gt; header = T): incomplete final line found by readTableHeader on &#39;C:/</span></a>
<a class="sourceLine" id="cb2-82" title="82"><span class="co">#&gt; Users/Administrator.MS-20180914RLKI/AppData/Local/Temp/RtmpWK9Cqj/</span></a>
<a class="sourceLine" id="cb2-83" title="83"><span class="co">#&gt; Rinst3c9c32f11d2a/GeoTcgaData/extdata/tcga_cli/ats.org.TCGA-2Y-A9GS.xml&#39;</span></a>
<a class="sourceLine" id="cb2-84" title="84"><span class="co">#&gt;              V1   V2    V3</span></a>
<a class="sourceLine" id="cb2-85" title="85"><span class="co">#&gt; 1: TCGA-2V-A95S haha Alive</span></a>
<a class="sourceLine" id="cb2-86" title="86"><span class="co">#&gt; 2: TCGA-2Y-A9GT 1624  Dead</span></a>
<a class="sourceLine" id="cb2-87" title="87"><span class="co">#&gt; 3: TCGA-2Y-A9GS  724  Dead</span></a>
<a class="sourceLine" id="cb2-88" title="88"></a>
<a class="sourceLine" id="cb2-89" title="89"><span class="co">#&#39; Multiple genes symbols may correspond to a same id. Some people think </span></a>
<a class="sourceLine" id="cb2-90" title="90"><span class="co">#&#39; that the expression value of this id should be </span></a>
<a class="sourceLine" id="cb2-91" title="91"><span class="co">#&#39; given to each gene, and some people think that the expression value of</span></a>
<a class="sourceLine" id="cb2-92" title="92"><span class="co">#&#39; this id should be deleted. The result of rep1 is to assign the expression</span></a>
<a class="sourceLine" id="cb2-93" title="93"><span class="co">#&#39; of this id to each gene, and rep2 deletes the expression.</span></a>
<a class="sourceLine" id="cb2-94" title="94"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-95" title="95"><span class="co">#&#39; param file1 input file, a data.frame or a matrixg</span></a>
<a class="sourceLine" id="cb2-96" title="96"><span class="co">#&#39; param string a string,sep of the gene</span></a>
<a class="sourceLine" id="cb2-97" title="97"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-98" title="98"><span class="co">#&#39; return a data.frame,rep1 is to assign the expression</span></a>
<a class="sourceLine" id="cb2-99" title="99"><span class="co">#&#39; of this id to each gene, and rep2 deletes the expression.</span></a>
<a class="sourceLine" id="cb2-100" title="100"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-101" title="101"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-102" title="102">aa &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;MARCH1 /// MMA&quot;</span>,<span class="st">&quot;MARC1&quot;</span>,<span class="st">&quot;MARCH2 /// MARCH3&quot;</span>,</a>
<a class="sourceLine" id="cb2-103" title="103">        <span class="st">&quot;MARCH3 /// MARCH4&quot;</span>,<span class="st">&quot;MARCH1&quot;</span>)</a>
<a class="sourceLine" id="cb2-104" title="104">bb &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;2.969058399&quot;</span>,<span class="st">&quot;4.722410064&quot;</span>,<span class="st">&quot;8.165514853&quot;</span>,<span class="st">&quot;8.24243893&quot;</span>,<span class="st">&quot;8.60815086&quot;</span>)</a>
<a class="sourceLine" id="cb2-105" title="105">cc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3.969058399&quot;</span>,<span class="st">&quot;5.722410064&quot;</span>,<span class="st">&quot;7.165514853&quot;</span>,<span class="st">&quot;6.24243893&quot;</span>,<span class="st">&quot;7.60815086&quot;</span>)</a>
<a class="sourceLine" id="cb2-106" title="106">input_fil &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">aa=</span>aa,<span class="dt">bb=</span>bb,<span class="dt">cc=</span>cc)</a>
<a class="sourceLine" id="cb2-107" title="107">rep1_result &lt;-<span class="st"> </span><span class="kw">rep1</span>(input_fil,<span class="st">&quot; /// &quot;</span>)</a>
<a class="sourceLine" id="cb2-108" title="108">rep1_result &lt;-<span class="st"> </span><span class="kw">rep2</span>(input_fil,<span class="st">&quot; /// &quot;</span>)</a>
<a class="sourceLine" id="cb2-109" title="109"></a>
<a class="sourceLine" id="cb2-110" title="110"><span class="co">#&#39; Convert  ENSEMBL gene id to gene Symbol in TCGA</span></a>
<a class="sourceLine" id="cb2-111" title="111"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-112" title="112"><span class="co">#&#39; param profile a data.frame</span></a>
<a class="sourceLine" id="cb2-113" title="113"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-114" title="114"><span class="co">#&#39; @return a data.frame, gene symbols and their expression value</span></a>
<a class="sourceLine" id="cb2-115" title="115"><span class="co">#&#39;</span></a>
<a class="sourceLine" id="cb2-116" title="116"><span class="co">#&#39; examples</span></a>
<a class="sourceLine" id="cb2-117" title="117">result &lt;-<span class="st"> </span><span class="kw">id_conversion</span>(profile)</a></code></pre></div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
